### PR TITLE
Update nsswitch.conf to use systemd-resolved

### PR DESCRIPTION
**Pull request by f0o**
# Update nsswitch.conf to use systemd-resolved

Partly related to https://github.com/kinvolk/Flatcar/issues/285 as suggested by @pothos 
This allows nsswitch to use systemd-resolved

# How to use

Update the nsswitch.conf ;)

# Testing done

```
#/etc/systemd/resolved.conf:

[Resolve]
DNS=10.0.0.2
Domains=~example.internal
```
```
# ping web.example.internal
PING web.example.internal (10.0.154.231) 56(84) bytes of data.
...
# ping google.com
PING google.com (172.217.21.142) 56(84) bytes of data.
...
```

This test shows that systemd's split-horizon (routed-domains) is working and nsswitch is also using systemd-resolved instead of /etc/resolv.conf (the DNS referenced is NOT recursive)

__HOWEVER__

it's important to note that anything using /etc/resolv.conf will fail to use systemd-resolved until the stublistener is re-enabled and the symlink it set to `/run/systemd/resolve/stub-resolv.conf` as outlined in the linked issue. (This limitation does not impose any regression as it didnt work before at all, so this is now a half-working solution - full working is achieved by removing `/lib/systemd/resolved.conf.d/50-no-dns-listener.conf` and replacing /etc/resolv.conf)

### CODE REVIEW

File: baselayout/nsswitch.conf [hunk 1]
```diff
@@ -4,7 +4,7 @@ passwd:      files usrfiles sss systemd
 shadow:      files usrfiles sss
 group:       files usrfiles sss systemd
 
-hosts:       files usrfiles dns myhostname
+hosts:       resolve [!UNAVAIL=return] files usrfiles dns myhostname
```
Comment (pothos) [inline baselayout/nsswitch.conf]: The first to look up should still be files and usrfiles:
```suggestion
hosts:       files usrfiles resolve [!UNAVAIL=return] dns myhostname
```

Actually I find it strange that `myhostname` comes last because this is normally between `files (usrfiles)` and `resolve` (or between `resolve` and `dns`) but that's not part of this PR. Edit: `myhostname` being last is somehow recommended according to the docs.

Comment (f0o) [reply baselayout/nsswitch.conf]: I took https://www.freedesktop.org/software/systemd/man/nss-systemd.html as example and moved myhostname to the back as it was already the last.

The ordering with files first makes little sense as resolved will already include those in the lookup which is why it's suggested to be first and have a quick return

Comment (pothos) [reply baselayout/nsswitch.conf]: I think the situation with `usrfiles` (nss-altfiles) may not be covered by `resolve`, that's why I suggested to move it just before `dns`.

Comment (f0o) [reply baselayout/nsswitch.conf]: I was looking into usrfiles (https://github.com/kubic-project/libnss_usrfiles ?) and it doesnt seem to care about hosts at all and only deals with user/group... In which case we can scrap it and put resolve back to the front to enable files caching

Comment (pothos) [reply baselayout/nsswitch.conf]: It's this here https://github.com/kinvolk/nss-altfiles and supports looking up in `/usr/share/baselayout/hosts`.

Comment (f0o) [reply baselayout/nsswitch.conf]: Okay but according to that documentation it should be `altfiles` and not `usrfiles`

Comment (pothos) [reply baselayout/nsswitch.conf]: Yes, a bit confusing, it's compiled with the option `--with-module-name=usrfiles`.

Comment (f0o) [reply baselayout/nsswitch.conf]: Ah ok, yeah slightly confusing ;)

I guess I mark this as resolved again 👍

Comment (pothos) [reply baselayout/nsswitch.conf]: Good, I'm running a test with all changes combined, thanks again for the PRs

Comment (pothos) [reply baselayout/nsswitch.conf]: There are test failures on Azure (the rest is fine except for silencing a warning about a new listening socket that is detected), I'll have to reproduce them and see what's wrong.

Comment (f0o) [reply baselayout/nsswitch.conf]: Which failures? happy to help

The new socket is to be expected due to the stub listener being enabled now

Comment (pothos) [reply baselayout/nsswitch.conf]: Eventually it works but systemd-resolved tries DNSSEC first and that fails.

Comment (f0o) [reply baselayout/nsswitch.conf]: Define fails, DNSSEC is preferred but defaults to allow-downgrade which should transparently retry without DNSSEC. Queries should only fail if your recursive DNS server sends bogus DNSSEC responses at which point the local resolver has to assume that it's a forged response and drop the answer accordingly.

Comment (pothos) [reply baselayout/nsswitch.conf]: It failed to resolve because DNSSEC is not supported and the retry was not transparent but failed the whole resolving action. I think DNSSEC should be opt-in until this behavior gets better.

File: baselayout/nsswitch.conf [hunk 2]
```diff
@@ -4,7 +4,7 @@ passwd:      files usrfiles sss systemd
 shadow:      files usrfiles sss
 group:       files usrfiles sss systemd
 
-hosts:       files usrfiles dns myhostname
+hosts:       files usrfiles resolve [!UNAVAIL=return] dns myhostname
```
Comment (pothos) [inline baselayout/nsswitch.conf]: This is order is what Fedora uses:
```suggestion
hosts:       files usrfiles resolve [!UNAVAIL=return] myhostname dns
```
I think it makes sense to try to resolve the hostname first but anyway, by default resolved would be used and covers this.

**Review by pothos — APPROVED** [review]
